### PR TITLE
Fix crash when window gets minimized

### PR DESCRIPTION
**Pull request by MeirBon**
Implements a simple check of whether the window dimensions are valid. If the window dimensions are invalid, e.g. when the window is minimized, render calls are not done. The game state is updated regardless.

This fixes https://github.com/iceiix/stevenarella/issues/275

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,646 @@ objects
...
+# BeatPulse healthcheck temp database
+healthchecksdb
+
+# Backup folder for Package Reference Convert tool in Visual Studio 2017
+MigrationBackup/
+
+# Ionide (cross platform F# VS Code tools) working folder
+.ionide/
+
+# End of https://www.toptal.com/developers/gitignore/api/rust,c++,visualstudiocode,visualstudio,clion,jetbrains,windows,macos,linux
```
Comment (iceiix) [inline .gitignore:660]: This .gitignore change would be better as a separate pull request as its not directly related to the issue

File: src/main.rs [hunk 2]
```diff
@@ -324,7 +324,6 @@ fn main2() {
         let delta = (diff.subsec_nanos() as f64) / frame_time;
         let (width, height) = window.window().inner_size().into();
         let (physical_width, physical_height) = window.window().inner_size().to_physical(game.dpi_factor).into();
-
```
Comment (iceiix) [inline src/main.rs:327]: Unnecessary whitespace change

File: src/main.rs [hunk 3]
```diff
@@ -363,7 +367,6 @@ fn main2() {
         ui_container.tick(&mut game.renderer, delta, width as f64, height as f64);
         game.renderer.tick(&mut game.server.world, delta, width, height, physical_width, physical_height);
 
-
```
Comment (iceiix) [inline src/main.rs:366]: Unnecessary whitespace change

File: src/render/mod.rs [hunk 4]
```diff
@@ -259,10 +259,13 @@ impl Renderer {
...
 
+            let fov = cgmath::Rad::from(cgmath::Deg(90.0_f32));
+            let aspect_ratio = (width as f32 / height as f32).max(1.0);
+
             self.perspective_matrix = cgmath::Matrix4::from(
                 cgmath::PerspectiveFov {
-                    fovy: cgmath::Rad::from(cgmath::Deg(90f32)),
-                    aspect: (width as f32 / height as f32),
+                    fovy: fov,
+                    aspect: aspect_ratio,
```
Comment (iceiix) [inline src/render/mod.rs:268]: If you name these variables `fovy` and `aspect`, then instead of fovy: fov, aspect: aspect_ratio you can write this more concisely as:

```rust
                cgmath::PerspectiveFov {
                      fovy,
                      aspect,
```

File: .gitignore (hunk 1 above)
Comment (MeirBon) [reply .gitignore:660]: Am making a new pr now.
